lib/Controller/WopiController.php

```diff
@@ -711,6 +739,82 @@ public function putRelativeFile($fileId,
 		}
 	}
 
+	private function lock(Wopi $wopi, string $lock): JSONResponse {
+		try {
+			$lock = $this->lockManager->lock(new LockContext(
+				$this->getFileForWopiToken($wopi),
+				ILock::TYPE_APP,
+				Application::APPNAME
+			));
+			return new JSONResponse();
+		} catch (NoLockProviderException|PreConditionNotMetException $e) {
+			return new JSONResponse([], Http::STATUS_BAD_REQUEST);
+		} catch (OwnerLockedException $e) {
+			return new JSONResponse([], Http::STATUS_LOCKED);
+		} catch (\Exception $e) {
+			return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+	private function unlock(Wopi $wopi, string $lock): JSONResponse {
+		try {
+			$this->lockManager->unlock(new LockContext(
+				$this->getFileForWopiToken($wopi),
+				ILock::TYPE_APP,
+				Application::APPNAME
+			));
+			return new JSONResponse();
+		} catch (NoLockProviderException|PreConditionNotMetException $e) {
+			return new JSONResponse([], Http::STATUS_BAD_REQUEST);
+		} catch (\Exception $e) {
+			return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	private function refreshLock(Wopi $wopi, string $lock): JSONResponse {
+		try {
+			$lock = $this->lockManager->lock(new LockContext(
+				$this->getFileForWopiToken($wopi),
+				ILock::TYPE_APP,
+				Application::APPNAME
+			));
+			return new JSONResponse();
+		} catch (NoLockProviderException|PreConditionNotMetException $e) {
+			return new JSONResponse([], Http::STATUS_BAD_REQUEST);
+		} catch (OwnerLockedException $e) {
+			return new JSONResponse([], Http::STATUS_LOCKED);
+		} catch (\Exception $e) {
+			return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	private function getLock(Wopi $wopi, string $lock): JSONResponse {
+		$locks = $this->lockManager->getLocks($wopi->getFileid());
+		return new JSONResponse();
+	}
+
+	/**
+	 * @throws NotFoundException
+	 * @throws GenericFileException
+	 * @throws LockedException
+	 * @throws ShareNotFound
+	 */
+	protected function wrappedFilesystemOperation(Wopi $wopi, callable $filesystemOperation): void {
+		$retryOperation = function () use ($filesystemOperation) {
+			$this->retryOperation(function () use ($filesystemOperation) {
```

Comment:
Is it possible to pass the callable directly?

```php
$this->retryOperation($filesystemOperation);
```

Reply:
Yes, good catch, addressed